pkg/dhpsi/dhpsi.go

```diff
@@ -49,9 +51,11 @@ func NewDeriveMultiplyShuffler(w io.Writer, n int64, gr Ristretto) (*DeriveMulti
 	if err := binary.Write(w, binary.BigEndian, &n); err != nil {
 		return nil, err
 	}
-	// and create the encoder
-	return &DeriveMultiplyShuffler{w: w, max: n, gr: gr, permutations: initP(n), b: make([][EncodedLen]byte, n)}, nil
-
+	// create the permutations
+	p, _ := permutations.NewKensler(n)
+	// and create the buffer map & encoder
+	b := make(map[int64][EncodedLen]byte, int(float64(n)*0.75))
```

Comment:
Is there a reason for using 3/4 of the size?

Reply:
during testing I noticed it tended to fill 75% of the map but its a very thin link. Maybe half or full would be better

Reply:
I see, but do you need this optimization for the Parallel version as well?
also, maybe just add an one line comment to state that.

Reply:
the parallel version doesn't implement caching at this time